applications/ContactStructuralMechanicsApplication/custom_utilities/derivatives_utilities.h

```diff
@@ -262,6 +262,69 @@ class DerivativesUtilities
         return delta_normal;
     }
 
+    /**
+     * @brief This method is used to compute the local increment of the normal
+     * @param rJacobian The jacobian on the GP
+     * @param rDNDe The local gradient
+     * @return The matrix containing the delta normals
+     * @note Not the mean, look in the contact utilities
+     * @note Hardcopied for performance
+     */
+    static inline array_1d<array_1d<double, 3>, TDim * TNumNodesMaster> GPDeltaNormalMaster(
```

Comment:
you sure you want to inline such a large function?

Reply:
It is not used in many places, but I can remove the inline in a future PR